packages/rolldown/src/plugin/plugin-context.ts

```diff
@@ -184,6 +184,8 @@ export class PluginContextImpl extends MinimalPluginContextImpl {
         )
         : res.external,
       ...info,
+      moduleSideEffects: info.moduleSideEffects ?? res.moduleSideEffects ??
+        null,
```

Comment:
I'm not sure about the merge here. There aren't much information about what `info.moduleSideEffects` is and how that should be handled.

Reply:
I don't understand how module option is. If you identify which has higher priority, I think the current merge logic is fine.